.github/workflows/update-release-version-template.yaml

```diff
@@ -0,0 +1,28 @@
+---
+name: Reusable Template to Update Release
+
+on:
```

Comment:
@ivanvc @jmhbnz I am not an expert but wondering shouldn't we use `workflow_dispatch:` here as well? With `write` permission, we need to be careful as much as possible. I am sure you have thought about security but still wanted to double check. Thanks! cc @ahrtr @serathius @wenjiaswe

Reply:
Agree we should use `contents: write` sparingly as there are security considerations.

We have `workflow_dispatch` on the scheduled job that calls this one in order to support running a workflow on demand https://github.com/etcd-io/website/pull/845/files#diff-709e89ffb8e7b46a9736507dd8f21a544b583f4fbc0a1fb01c009f3b41c9178cR9.

@ivanvc if we are using this action to raise a pull request automatically for the version update can we get away with `pull-requests: write` or do we absolutely need `contents: write` as well?

Reply:
I want to test this first and will get back to you with the answer. Either way, I'll narrow down the permissions. Good catch :).

As per `workflow_dispatch:`, I'm fine removing it. It's just a convenience to run them on-demand, as @jmhbnz stated. Also, these triggers can only be done by people with write access to the repo (ref: [GitHub docs](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow#configuring-a-workflow-to-run-manually)).

Reply:
Without the `contents: write` permission, I get a 403 when pushing to the repository :face_with_head_bandage:. Refer to this run without `contents: write`: https://github.com/ivanvc/etcd-website/actions/runs/9310254830/job/25627319064.

Reply:
I tried some other options without pushing to a branch. However, if I have the local commit, GitHub's CLI still asks to push it to a remote.

An alternative would be to keep this action running in a fork, and it opens a PR targeting this repository. That way, we won't need the action with `contents: write` in this repository.

Please let me know what you think.

Reply:
Thanks much @ivanvc !! If we have control on merging a PR, that would be great. Also, if I understood correctly, `workflow_dispatch:` is a good security practice to manually approve the workflow. So, I didn't say to remove it but I was suggesting/checking that we should probably add it in [update-release-version-template.yaml](https://github.com/etcd-io/website/pull/845/files#diff-55aabdf214faf6ad31817739890a15aa1bf9b21f98f313dd670d605207fdca92) as well.

Reply:
@jmhbnz, do you have any thoughts on:

> An alternative would be to keep this action running in a fork, and it opens a PR targeting this repository. That way, we won't need the action with contents: write in this repository.

It could remove the security issue that the action has write access to the repository. On the other hand, it might not be ideal that the action doesn't live in the etcd-io organization.

Reply:
Hey Team - I am happy to forge ahead and merge (provided @spzala agrees) for the following reasons:
- This workflow does not run `on: pull_request` meaning it only runs against code we have already merged, and only on our schedule or if we manually trigger it (requires maintainer to do so).
- We will be manually approving and deciding to merge any pull requests it raises
- Any changes to the behaviour of this workflow will be reviewed by our pull request process (and should be closely scrutinized).

Reply:
@jmhbnz +1 sounds great. Thank you and @ivanvc so much!